engine/src/test/java/io/camunda/zeebe/engine/state/migration/JobBackoffRestoreMigrationTest.java

```diff
@@ -0,0 +1,145 @@
+  private ColumnFamily<DbCompositeKey<DbLong, DbForeignKey<DbLong>>, DbNil> backoffColumnFamily;
+  private final JobStateValue jobState = new JobStateValue();
+  private ColumnFamily<DbForeignKey<DbLong>, JobStateValue> statesJobColumnFamily;
+
+  @BeforeEach
+  public void setup() {
+    jobKey = new DbLong();
+    final DbForeignKey<DbLong> fkJob = new DbForeignKey<>(jobKey, ZbColumnFamilies.JOBS);
+    jobsColumnFamily =
+        zeebeDb.createColumnFamily(
+            ZbColumnFamilies.JOBS, transactionContext, jobKey, jobRecordToRead);
+
+    statesJobColumnFamily =
+        zeebeDb.createColumnFamily(
+            ZbColumnFamilies.JOB_STATES, transactionContext, fkJob, jobState);
+
+    backoffKey = new DbLong();
+    backoffJobKey = new DbCompositeKey<>(backoffKey, fkJob);
+    backoffColumnFamily =
+        zeebeDb.createColumnFamily(
+            ZbColumnFamilies.JOB_BACKOFF, transactionContext, backoffJobKey, DbNil.INSTANCE);
+
+    jobKey.wrapLong(1);
+  }
+
+  // regression test of https://github.com/camunda/zeebe/issues/14329
+  @Test
+  public void shouldRestoreIfBackoffColumnIsEmpty() {
+    // given
+    final MutableJobState jobState = processingState.getJobState();
+    final JobRecord record = createJobRecord(1000);
+    jobState.create(jobKey.getValue(), record);
+    jobState.fail(jobKey.getValue(), record);
+    backoffKey.wrapLong(record.getRecurringTime());
+    backoffColumnFamily.deleteExisting(backoffJobKey);
+
+    // when
+    assertThat(jobBackoffRestoreMigration.needsToRun(processingState)).isTrue();
+    jobBackoffRestoreMigration.runMigration(processingState);
+
+    // then
+    assertThat(backoffColumnFamily.isEmpty()).isFalse();
+  }
+
+  // regression test of https://github.com/camunda/zeebe/issues/14329
+  @Test
+  public void shouldRestoreIfFailedJobsAreMoreThanBackoffJob() {
+    // given
+    final MutableJobState jobState = processingState.getJobState();
+    final JobRecord record = createJobRecord(1000);
+    jobState.create(jobKey.getValue(), record);
+    jobState.fail(jobKey.getValue(), record);
+    backoffKey.wrapLong(record.getRecurringTime());
+    backoffColumnFamily.deleteExisting(backoffJobKey);
+
+    jobKey.wrapLong(2);
+    final JobRecord backoffRecord = createJobRecord(2000);
+    jobState.create(jobKey.getValue(), backoffRecord);
+    jobState.fail(jobKey.getValue(), backoffRecord);
+    backoffKey.wrapLong(backoffRecord.getRecurringTime());
+    assertThat(backoffColumnFamily.count()).isEqualTo(1);
+
+    // when
+    assertThat(jobBackoffRestoreMigration.needsToRun(processingState)).isTrue();
+    jobBackoffRestoreMigration.runMigration(processingState);
+
+    // then
+    assertThat(backoffColumnFamily.isEmpty()).isFalse();
+    assertThat(backoffColumnFamily.count()).isEqualTo(2);
+  }
+
+  // regression test of https://github.com/camunda/zeebe/issues/14329
+  @Test
+  public void shouldDoNothingIfFailedJobsAreTheSameOfBackoff() {
```

Comment:
:upside_down_face: Is the following a typo?
```suggestion
  public void shouldDoNothingIfFailedJobsAreTheSameAsBackoff() {
```

Reply:
yes 😄